### PR TITLE
Adding a .NET 8 target follow up for extensions packages

### PR DESCRIPTION
**Pull request by m-redding**
A follow-up to https://github.com/Azure/azure-sdk-for-net/pull/46637

A few errors in extensions packages were missed by the CIs. Fixing those here.

### CODE REVIEW

File: sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventBinding.cs [hunk 1]
```diff
@@ -86,8 +86,13 @@ private static IReadOnlyDictionary<string, Type> GetBindingDataContract(Paramete
         public async Task<ITriggerData> BindAsync(object value, ValueBindingContext context)
         {
             var request = (HttpRequestMessage)value;
+#if NET8_0_OR_GREATER
```
Comment (m-redding) [inline sdk/entra/Microsoft.Azure.WebJobs.Extensions.AuthenticationEvents/src/AuthenticationEventBinding.cs:89]: [HttpRequestMessage.Properties](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httprequestmessage.properties?view=net-8.0) is now obsolete. The guidance is to use [HttpRequestMessage.Options](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httprequestmessage.options?view=net-8.0) instead. However, the Options property is not available in netstandard2.0.

**Comment by m-redding** [comment]
From the pipeline history, it looks like the failures in net - storage - functions - tests are unrelated to both this change and the change in https://github.com/Azure/azure-sdk-for-net/pull/46637

**Review by jsquire — APPROVED** [review]

**Comment by m-redding** [comment]
Override failures per my comment above:
> From the pipeline history, it looks like the failures in net - storage - functions - tests are unrelated to both this change and the change in https://github.com/Azure/azure-sdk-for-net/pull/46637

**Pull request merged**